daemon/logger/jsonfilelog/read.go

```diff
@@ -171,9 +178,21 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 	}
 	defer func() {
 		f.Close()
+		fileWatcher.Remove(name)
 		fileWatcher.Close()
 	}()
 
+	logWatcherCtx, logWatcherClosed := context.WithCancel(ctx)
```

Comment:
Other than multiple owners of `cancel()`? It's probably ok. I prefer the separation, but assuming nothing panics it seems ok.